plugins/statistics.koplugin/main.lua

```diff
@@ -1074,6 +1074,52 @@ The max value ensures a page you stay on for a long time (because you fell aslee
                         end,
                         separator = true,
                     },
+                    {
+                        text_func = function()
+                            -- @translators %1 is the time in the format 00:00
+                            return T(_("Calendar days start at %1 "),
+                                string.format("%02d:%02d", self.settings.calendar_day_start_hour or 0,
+                                                           self.settings.calendar_day_start_minute or 0)
+                            )
+                        end,
+                        callback = function(touchmenu_instance)
+                            local DateTimeWidget = require("ui/widget/datetimewidget")
+                            local start_of_day_widget = DateTimeWidget:new{
+                                hour = self.settings.calendar_day_start_hour or 0,
+                                min = self.settings.calendar_day_start_minute or 0,
+                                hour_max = 6,
+                                ok_text = _("Set time"),
+                                title_text = _("Set starting time of a day"),
+                                info_text =_([[
+Set the time when the day timeline should start.
+
+If you read past midnight, and wish this reading to appear with the previous evening's reading, you can set this to 04:00 for example.
+
+Time is in hours and minutes.]]),
+                                callback = function(time)
+                                    if time.hour and time.min then
+                                        self.settings.calendar_day_start_hour = time.hour
+                                        self.settings.calendar_day_start_minute = time.min
+                                        touchmenu_instance:updateItems()
+                                    else
+                                        UIManager:show(InfoMessage:new{
+                                            text = _("Time couldn't be set"),
```

Comment:
Can this happen ? I guess this InfoMessage comes from the "Set date/time", which can fail for hardware/os reasons.

Reply:
Removed it.